File: partiql-ast/src/main/java/org/partiql/ast/v1/SelectStar.java
package org.partiql.ast.v1;

import lombok.Builder;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Collection;
import java.util.Collections;

/**
 * TODO docs, equals, hashcode
 */
@Builder(builderClassName = "Builder")
public class SelectStar extends Select {
    @Nullable
    public final SetQuantifier setq;

    public SelectStar(@Nullable SetQuantifier setq) {
        this.setq = setq;
    }

    @NotNull
    @Override
    public Collection<AstNode> children() {
        return Collections.emptyList();
    }

    @Override
    public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
        return visitor.visitSelectStar(this, ctx);
    }
}

File: partiql-ast/src/main/kotlin/org/partiql/ast/v1/SelectStar.kt
package org.partiql.ast.v1

/**
 * TODO docs, equals, hashcode
 */
public class SelectStar(
    @JvmField
    public var setq: SetQuantifier?,
) : Select() {
    public override fun children(): Collection<AstNode> = emptyList()

    public override fun <R, C> accept(visitor: AstVisitor<R, C>, ctx: C): R =
        visitor.visitSelectStar(this, ctx)
}
